tests/type_inference/type_inference_visitor_unittest.py

```diff
@@ -0,0 +1,305 @@
+
+    def test_const_bool(self):
+        """testing if the function set_const has the correct output for node
+        type boolean."""
+        module = astroid.parse("""True""")
+        type_visitor.visit(module)
+        expected = bool
+        self.assertEqual(expected, module.body[0].value.type_constraints)
+
+    def test_const_none(self):
+        """testing if the function set_const has the correct output for
+        NoneType node."""
+        module = astroid.parse("""None""")
+        type_visitor.visit(module)
+        expected = type(None)
+        self.assertEqual(expected, module.body[0].value.type_constraints)
+
+
+class TypeInferenceVisitorTest(unittest.TestCase):
+    """testers for type_inference_visitor. Modules are been passed in instead
+    of single nodes."""
+
+    def test_binop_same_type_operands(self):
+        """testing if a binary operator passed into TypeVisitor
+        has the correct type_constraints attribute when both operands have
+        the same type.
+
+        Note: astroid.parse() and type_visitor both have a return type
+        Module(), if we directly look for nodes under Module(), the node
+        binop will not be returned, but instead, it will be a wrapper for
+        binop -> astroid.Expr. To find the binop wrapped by the Expr node,
+        we need to use Expr.value. So binop is 2 layers under the return
+        type of astroid.parse and type_visitor.visit (binop is 2 layers
+        under module).
+
+        An example from AST:
+        > parseprint('-a')
+        Module(body=[
+        Expr(value=UnaryOp(op=USub(), operand=Name(id='a', ctx=Load()))),
+        ])
+        """
+        module = astroid.parse("""10 + 2""")    # int
+        visited_module = type_visitor.visit(module)
+        self.assertEqual(int, visited_module.body[0].value.type_constraints)
+
+    def test_binop_diff_type_operands(self):
+        """testing if a binary operator that's been passed into
+        TypeVisitor has the correct type_constraints attribute
+        when operands have different types.
+        """
+        module = astroid.parse("""6 + 0.3""")   # float
+        visited_module = type_visitor.visit(module)
+        self.assertEqual(float, visited_module.body[0].value.type_constraints)
+
+    def test_unary(self):
+        """testing if a unary operator that's been passed into
+        TypeVisitor has the correct type_constraints attribute.
+        """
+        module = astroid.parse("""-2""")    # int
+        visited_module = type_visitor.visit(module)
+        self.assertEqual(int, visited_module.body[0].value.type_constraints)
+
+    def test_list_same_type_elements(self):
+        """testing if a list that's been passed into TypeVisitor
+        has the correct type_constraints attribute.
+
+        The list contains only 1 type of elements.
+        """
+        module = astroid.parse("""['hi', 'how', 'is', 'life']""")  # List(str)
```

Comment:
I don't think you need the comment; it's quite clear what the expected type is from the `assertEqual` line.